cmd/opm/alpha/veneer/semver.go

```diff
@@ -18,17 +18,29 @@ import (
 func newSemverCmd() *cobra.Command {
 	output := ""
 	cmd := &cobra.Command{
-		Use:   "semver <filename>",
-		Short: "Generate a file-based catalog from a single 'semver veneer' file",
-		Long:  `Generate a file-based catalog from a single 'semver veneer' file`,
-		Args:  cobra.MinimumNArgs(1),
+		Use:   "semver [FILE]",
+		Short: "Generate a file-based catalog from a single 'semver veneer' file \nWhen FILE is '-' or not provided, the veneer is read from standard input",
+		Long:  "Generate a file-based catalog from a single 'semver veneer' file \nWhen FILE is '-' or not provided, the veneer is read from standard input",
+		Args:  cobra.MaximumNArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			ref := args[0]
-
 			var (
-				err error
+				data io.Reader
+				err  error
 			)
 
+			// Handle different input argument types
+			if len(args) == 0 || args[0] == "-" {
```

Comment:
IntelliJ/GoLand offers this method extraction (naming is mine). I think with good function naming, people can just read right over it, as it's a utility function:
```suggestion
			data, err, err2 := openFileOrPipe(cmd, args, data, err)
			if err2 != nil {
				return err2
			}
		
func openFileOrPipe(cmd *cobra.Command, args []string, data io.Reader, err error) (io.Reader, error, error) {
	if len(args) == 0 || args[0] == "-" {
		// When no arguments or "-" is passed to the command,
		// assume input is coming from stdin
		data = cmd.InOrStdin()
	} else {
		// Otherwise open the file passed to the command
		data, err = os.Open(args[0])
		if err != nil {
			return nil, nil, err
		}
	}
	return data, err, nil
}	
			
```